File: task1/src/observer.py
#!/usr/bin/env python

import rospy
from geometry_msgs.msg import Point

class Observer():
    def __init__(self):
        self.sub = rospy.Subscriber('point', Point, self.callback)

    def callback(self, point):
        rospy.loginfo("The runner's position is: x = [%f], y = [%f], z = [%f]", point.x, point.y, point.z)


if __name__ == '__main__':
    rospy.init_node('observer_task1')
    Observer()
    rospy.spin()

File: task1/src/observer.cpp
#include "ros/ros.h"
#include "geometry_msgs/Point.h"

void chatterCallback(const geometry_msgs::Point::ConstPtr& point){
 ROS_INFO("The coordinates are: x = [%f], y = [%f], z = [%f]", point->x, point->y, point->z);
}

int main(int argc, char **argv){
 ros::init(argc, argv, "observer");
 ros::NodeHandle n;
 ros::Subscriber sub = n.subscribe("point", 1000, chatterCallback);
 ros::spin();

 return 0;

}
